ruby/ql/lib/codeql/ruby/dataflow/internal/DataFlowPublic.qll

```diff
@@ -60,6 +60,16 @@ class CallNode extends LocalSourceNode {
   Node getKeywordArgument(string name) { result.asExpr() = node.getKeywordArgument(name) }
 }
 
+/** A data-flow node corresponding to a method call in the control-flow graph. */
+class MethodCallNode extends CallNode {
+  private CfgNodes::ExprNodes::MethodCallCfgNode node;
```

Comment:
Make the super class field non-private and

```suggestion
  override CfgNodes::ExprNodes::MethodCallCfgNode node;
```
Alternatively, just add the `getMethodName` method directly to `CallNode`. I think it is fine for that method to return `none()` in case the underlying Call is not a MethodCall.

Reply:
I like the idea of moving `getMethodName` into `CallNode`, as the predicate name implies that we are interested in method calls.